resources/airflow.rb

```diff
@@ -1708,36 +1710,34 @@ def self.create_infil_mech_vent_objects(model, building, infil, mech_vent, wind_
 
       infil_program.addLine("If #{mech_vent.cfis_t_sum_open_var.name} < CFIS_t_min_hr_open")
       infil_program.addLine("  Set CFIS_t_fan_on = 60 - (CFIS_t_min_hr_open - #{mech_vent.cfis_t_sum_open_var.name})") # minute at which the blower needs to turn on to meet the ventilation requirements
-      infil_program.addLine("  If ((Minute+0.00001) >= CFIS_t_fan_on) || #{mech_vent.cfis_on_for_hour_var.name}")
-
-      # Supply fan needs to run for remainder of hour to achieve target minutes per hour of operation
-      infil_program.addLine("    If #{mech_vent.cfis_on_for_hour_var.name}")
-      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = 1")
-      infil_program.addLine("    Else")
-      infil_program.addLine("      Set cfistemp1 = (@Mod (60.0-CFIS_t_fan_on) (60.0*ZoneTimeStep))")
-      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = cfistemp1/(60.0*ZoneTimeStep)") # calculates the portion of the current timestep the CFIS damper needs to be open
-      infil_program.addLine("      Set #{mech_vent.cfis_on_for_hour_var.name} = 1") # CFIS damper will need to open for all the remaining timesteps in this hour
-      infil_program.addLine("    EndIf")
+      # Evaluate condition of if supply fan have to run to achieve target minutes per hour of operation
+      infil_program.addLine("  If (Minute+0.00001) >= CFIS_t_fan_on")
+      infil_program.addLine("    Set #{mech_vent.cfis_on_for_hour_var.name} = 1") # on flag for duct program
+      # Consider fan rtf read in current calling point (results of previous time step) + CFIS_t_fan_on based on min/hr requirement and previous EMS results.
+      infil_program.addLine("    Set cfis_fan_runtime1 = @Max (@ABS(Minute - CFIS_t_fan_on)) (fan_rtf_hvac * ZoneTimeStep * 60)")
```

Comment:
The use of `cfis_fan_runtime`, `cfis_fan_runtime1`, and `cfis_fan_runtime2` is a bit confusing. Is it possible to use better names or consolidate them?

Reply:
Lol sorry for that, I was trying to keep the final cfis_fan_runtime easier to track in edd so I made the process variables whatever names they can be. Will it be better to overwrite them with the same name?

Reply:
That seems reasonable to me.